### PR TITLE
(refactor): Modify helpers to fix unnecessary field

### LINKED ISSUE #1: Fields creates unecessary "field"
## Description
```json
    {
      "title": "This is text",
      "description": "This is text",
      "url": "https://google.com",
      "color": 12341234,
      "fields": [
        {
          "name": "This is text",
          "value": "https://google.com"
        }
      ]
    }
```

The above input results in the following invalid output
```go
{{ $embed := cembed ( sdict

"title" "This is text"
"description" "This is text"
"url" "https://google.com"
"color" 12341234
"fields" ( cslice 
	"fields" ( sdict  <---- extra "fields"
		"name" "This is text" 
		"value" "https://google.com"
	)
	) )
}}
```

Need to remove the command creator that adds the extra fields. we should evaluate each `sdict` in its own parens as each item in the `cslice`

## Expected Output

```go
{{ $embed := cembed ( sdict
  "title" "This is text"
  "description" "This is text"
  "url" "https://google.com"
  "color" 12341234
  "fields" ( cslice 
    ( sdict 
		"name" "This is text" 
		"value" "https://google.com"
    )
  )
)
}}
```

Yagbot docs: https://docs.yagpdb.xyz/reference/templates#custom-types

### LINKED ISSUE #2: Fields to not accept null
## Description
```json
    {
      "title": "This is text",
      "description": "This is text",
      "url": "https://google.com",
      "color": null
    }
```

Sometimes when creating an embed, we may not know the color right away, the outputted `null` should be a valid json.

## Expected output

Suggestion: replace `null` with `nil` per Go syntax. not 100% sure, but possible to try. Do we want to try `nil` or just SKIP over any fields that have `null` values? Open to trying both.

```go
{{ $embed := cembed ( sdict
"title" "This is text"
"description" "This is text"
"url" "https://google.com"
"color" nil )
}}
```

### PR DESCRIPTION
**Pull request by a-camarillo**
Fixes #1
Fixes #2 

Abstract helpers functions for readability, fix unnecessary field, and add function to handle color embeds

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -1,67 +1,94 @@
...
-  key: string,
-  args: string[] | { [key: string]: string },
-  type?: ICommand
+// Function for handling fields with object values
+export const handleObjectEmbed = (
+    field: string,
+    fieldValue: object,
+): string => {
+    if (field === "fields") {
+        let obj = '( sdict\n'
```
Comment (jwu910) [inline src/utils/helpers.ts]: can we rename `obj` to something more meaningful here?

Comment (jwu910) [reply src/utils/helpers.ts]: Possibly something like `commandStr`, `objectStr`, or `fieldObjectStr` to help identify what it conceptually is as well as what type we can expect it to be?

File: src/utils/helpers.ts [hunk 2]
```diff
@@ -1,67 +1,94 @@
...
+                case "color":
+                    return handleColorEmbed(key, embed[key]);
+                case "fields":
+                    return handleArrayEmbed(key, embed[key]);
+                case "author":
+                case "image":
+                case "footer":
+                case "thumbnail":
+                case "video":
+                case "provider":
```
Comment (jwu910) [inline src/utils/helpers.ts]: nice, can we sort these just for nitpick preference?

**Review by jwu910 — CHANGES_REQUESTED** [review]
Just a couple comments, overall definitely looking cleaner and easier to read. thanks for the PR

**Comment by a-camarillo** [comment]
alright hopefully those names are better and the case statements are now alphabetically sorted

**Review by jwu910 — APPROVED** [review]
Nice thanks

**Pull request merged**